### PR TITLE
record which jupyblog version was used upon rendering

### PR DESCRIPTION
**Pull request by fjborquez**
I tried to resolve issue #2

### CODE REVIEW

File: src/jupyblog/md.py [hunk 1]
```diff
@@ -263,6 +264,7 @@ def render(self, name, *, include_source_in_footer):
         md_ast = self.parser(md_raw)
         # TODO: replace and use model object
         metadata = parse_metadata(md_raw)
+        metadata['jupyblog']['version'] = version()
```
Comment (edublancas) [inline src/jupyblog/md.py:267]: looks good; please add the "version" under a `metadata` section. so it's more explicit that this is an automated field that should not be edited by users

**Review by edublancas — CHANGES_REQUESTED** [review]

File: src/jupyblog/md.py (hunk 1 above)
Comment (fjborquez) [reply src/jupyblog/md.py:267]: I will do it during the day! thanks!

File: src/jupyblog/md.py [hunk 2]
```diff
@@ -21,6 +22,7 @@
 JUPYBLOG = """\
 jupyblog:
   execute_code: false
+  version: 0.0.4
```
Comment (edublancas) [inline src/jupyblog/md.py:25]: users are not expected to put this manually so we can remove this line

File: src/jupyblog/md.py [hunk 3]
```diff
@@ -382,3 +383,18 @@ def run_snippets(md_ast, content, front_matter, img_dir, canonical_name):
...
 
     return md_out
+
+
+def version():
+    return jupyblog.__version__
+
+
+def set_version_to_metadata(metadata):
+    metadata['jupyblog']['version'] = version()
```
Comment (edublancas) [inline src/jupyblog/md.py:393]: we can simplify this and just assign 

```python
metadata['jupyblog']['version'] = jupyblog.__version__
```

File: src/jupyblog/models.py [hunk 4]
```diff
@@ -126,10 +126,14 @@ class Settings(BaseModel):
...
     execute_code : bool, default=True
         Execute code snippets.
+
+    version : str, default=''
+        Jupyblog version used to create the rendered file
     """
     serialize_images: bool = False
     allow_expand: bool = False
     execute_code: bool = True
+    version: str = ''
```
Comment (edublancas) [inline src/jupyblog/models.py:136]: let's put this under a `metadata` section, you'll have to create a new Metadata model and put version there, check pydantic's docs for details

**Review by edublancas — CHANGES_REQUESTED** [review]

**Comment by fjborquez** [comment]
Sorry. I will fix them after work!

**Comment by edublancas** [comment]
sure, no rush

**Comment by fjborquez** [comment]
Hey!, I tried to do my best but there are many code blocks that I did not understand well. Anyway, let me know again if I did something wrong :).

**Pull request closed without merging**